constraint/pkg/apis/templates/transform_test.go

```diff
@@ -0,0 +1,355 @@
+				},
+			},
+			exp: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &trueBool,
+				Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+					"message": {
+						Type: "string",
+					},
+					"labels": {
+						Type:                   "array",
+						XPreserveUnknownFields: &trueBool,
+						Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+							Schema: &apiextensionsv1beta1.JSONSchemaProps{
+								Type:                   "object",
+								XPreserveUnknownFields: &trueBool,
+								Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+									"message": {
+										Type: "string",
+									},
+									"labels": {
+										Type:                   "array",
+										XPreserveUnknownFields: &trueBool,
+										Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+											Schema: &apiextensionsv1beta1.JSONSchemaProps{
+												Type:                   "object",
+												XPreserveUnknownFields: &trueBool,
+												Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+													"key":          {Type: "string"},
+													"allowedRegex": {Type: "string"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Two deep Items",
+			v: &apiextensionsv1beta1.JSONSchemaProps{
+				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+					Schema: &apiextensionsv1beta1.JSONSchemaProps{
+						Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+							Schema: &apiextensionsv1beta1.JSONSchemaProps{
+								Type: "object",
+								Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+									"key":          {Type: "string"},
+									"allowedRegex": {Type: "string"},
+								},
+							},
+						},
+					},
+				},
+			},
+			exp: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &trueBool,
+				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+					Schema: &apiextensionsv1beta1.JSONSchemaProps{
+						XPreserveUnknownFields: &trueBool,
+						Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+							Schema: &apiextensionsv1beta1.JSONSchemaProps{
+								Type:                   "object",
+								XPreserveUnknownFields: &trueBool,
+								Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+									"key":          {Type: "string"},
+									"allowedRegex": {Type: "string"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Two deep AdditionalProperties",
+			v: &apiextensionsv1beta1.JSONSchemaProps{
+				AdditionalProperties: &apiextensionsv1beta1.JSONSchemaPropsOrBool{
+					Schema: &apiextensionsv1beta1.JSONSchemaProps{
+						AdditionalProperties: &apiextensionsv1beta1.JSONSchemaPropsOrBool{
+							Allows: false,
+						},
+					},
+				},
+			},
+			exp: &apiextensionsv1beta1.JSONSchemaProps{
+				XPreserveUnknownFields: &trueBool,
+				AdditionalProperties: &apiextensionsv1beta1.JSONSchemaPropsOrBool{
```

Comment:
Per the spec, AdditionalProperties just shouldn't be set:

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema

Reply:
We resolved this in [another thread](https://github.com/open-policy-agent/frameworks/pull/113#discussion_r626039744)